MagneticField/GeomBuilder/plugins/dd4hep/DD4hep_VolumeBasedMagneticFieldESProducerFromDB.cc

```diff
@@ -187,9 +189,10 @@ std::unique_ptr<MagneticField> DD4hep_VolumeBasedMagneticFieldESProducerFromDB::
       "<MaterialSection label=\"materials.xml\"><ElementaryMaterial name=\"materials:Vacuum\" density=\"1e-13*mg/cm3\" "
       "symbol=\" \" atomicWeight=\"1*g/mole\" atomicNumber=\"1\"/></MaterialSection>");
 
-  auto ddet = make_unique<cms::DDDetector>("cmsMagneticField:MAGF", sblob, true);
+  if (nullptr == detector_)
+    detector_ = new cms::DDDetector("cmsMagneticField:MAGF", sblob, true);
```

Comment:
now that the `new` is used instead of the unique pointer, I think the `detector_` needs to be `delete`-ed too, no?

Reply:
Done